### DISCUSSION TITLE
Internet Server - Websockets - Build Error "unknown type name 'httpd_ws_frame_t';"

### DISCUSSION THREAD

**Discussion by mikebeepboop** [Mair/esp32-course#78] [Course Qeustions] [answered]
Hi Everyone,
I can't figure our why my "idf.py build" is complaining about the not knowing what the "httpd_ws_frame_t" type is in the Websockets project :

************* From idf.py build output *************************
/main.c.obj -c ../main/main.c
../main/main.c: In function 'send_ws_message':
../main/main.c:65:3: error: unknown type name 'httpd_ws_frame_t'; did you mean 'httpd_work_fn_t'?
   httpd_ws_frame_t ws_message = {
   ^~~~~~~~~~~~~~~~
****************************************************************

My whole project in VSCode is green with no errors resolving functions/headers.

I have included the header that defines the "httpd_ws_frame_t" struct type : #include "esp_http_server.h"

I can also click through to the struct definition in the header file using CTRL + <click the "httpd_ws_frame_t" on line 65> to see its definition. I also re-wrote the struct definition manually and VSCode recognized all the members with auto-complete.

I'm using esp-idf 4.4

Anyone else have this issue, or have any ideas?

**Comment by Mair** [comment, marked as answer]
@mikebeepboop 
Did you add the web socket to the menuconfig https://learnesp32.com/videos/14/5_web%20sockets%20continued at 13:50?

**Reply by mikebeepboop** [reply]
Thanks, this was the missing piece. I had edited the menuconfig but somehow missed the most important check box :/
